cmd/query/app/query_parser.go

```diff
@@ -143,36 +187,154 @@ func (p *queryParser) parse(r *http.Request) (*traceQueryParameters, error) {
 	return traceQuery, nil
 }
 
-func (p *queryParser) parseTime(param string, r *http.Request) (time.Time, error) {
-	value := r.FormValue(param)
-	if value == "" {
-		if param == startTimeParam {
+// parseDependenciesQueryParams takes a request and constructs a model of dependencies query parameters.
+//
+// The dependencies API does not operate on the latency space, instead its timestamps are just time range selections,
+// and the typical backend granularity of those is on the order of 15min or more. As such, microseconds aren't
+// useful in this domain and milliseconds are sufficient for both times and durations.
+func (p *queryParser) parseDependenciesQueryParams(r *http.Request) (dqp dependenciesQueryParameters, err error) {
+	dqp.endTs, err = p.parseTime(r, endTsParam, time.Millisecond)
+	if err != nil {
+		return dqp, err
+	}
+
+	dqp.lookback, err = parseDuration(r, lookbackParam, durationUnitsParser{units: time.Millisecond}, defaultDependencyLookbackDuration)
+	return dqp, err
+}
+
+// parseMetricsQueryParams takes a request and constructs a model of metrics query parameters.
+//
+// Why the API is designed using an end time (endTs) and lookback:
```

Comment:
Also added reasons for various parameter decisions here. Again, if you can think of a better place to put these, I can move these.